data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/SqsWorker.java

```diff
@@ -452,4 +467,21 @@ private int getApproximateReceiveCount(final Message message) {
     void stop() {
         isStopped = true;
     }
+
+    public void recordSqsException(final AwsServiceException e) {
+        // AWS SQS emits some of their exceptions without the matching HTTP code. As we want to generate an aggregate version of
+        // these exceptions, we have to explicitly catch the type alongside the status code for the ones that leverage the status
+        // code (i.e. InvalidAddressException)
+        if (e.statusCode() == 403 ||
+                e instanceof KmsAccessDeniedException) {
+            sqsMessageAccessDeniedCounter.increment();
+        } else if (e.statusCode() == 404 ||
+                e instanceof QueueDoesNotExistException ||
+                e instanceof KmsNotFoundException) {
+            sqsQueueNotFoundCounter.increment();
```

Comment:
`e instanceof KmsNotFoundException` do we need a new counter `sqsKmsKeyNotFoundCounter` ?